s4e-web/src/app/utils.ts

```diff
@@ -12,3 +14,19 @@ export function activateMatcher(url: UrlSegment[]) {
   }
   return null;
 }
+
+export function adminDashboardMatcher(url: UrlSegment[]) {
+  const profileQuery: ProfileQuery = AppInjector.get(ProfileQuery);
+  profileQuery.ngOnInit();
+  const lastUrlParameters = url[0];
+  const isDashboardUrl = !!lastUrlParameters && lastUrlParameters.path === 'dashboard';
+  return profileQuery.isAdmin && isDashboardUrl ? ({consumed: [url[0]]}) : null;
+}
+
+export function managerDashboardMatcher(url: UrlSegment[]) {
+  const profileQuery: ProfileQuery = AppInjector.get(ProfileQuery);
+  profileQuery.ngOnInit();
+  const lastUrlParameters = url[0];
+  const isDashboardUrl = !!lastUrlParameters && lastUrlParameters.path === 'dashboard';
+  return profileQuery.isManager && !profileQuery.isAdmin && isDashboardUrl ? ({consumed: [url[0]]}) : null;
+}
```

Comment:
Are these needed? Couldn't router have overloaded paths?

Reply:
No, unfortunately it couldn't. 

1. Angular as a framework handle routing by going to the first path it achieves and then stops going through routes.
2. When routing is handled by Guard, it can be redirected to other sites, but will not continue to next (overloaded or not) path on rejection.
3. I'm not sure if in other versions than 7.x.x is it solved in another way, but in our one path and matcher can not be used together.